src/main/java/com/emc/pravega/perf/WriterWorker.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.emc.pravega.perf;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class WriterWorker extends Worker implements Callable<Void> {
+
+    WriterWorker(int sensorId, int eventsPerSec, int secondsToRun,
+                 boolean isRandomKey, int messageSize, Instant start,
+                 PerfStats stats, String streamName, long totalEvents) {
+
+        super(sensorId, eventsPerSec, secondsToRun,
+                isRandomKey, messageSize, start,
+                stats, streamName, totalEvents, 0);
+    }
+
+    public abstract CompletableFuture writeData(String key, String data);
+
+    public abstract void flush();
+
+    public abstract long eventCountIncrementAndGet();
+
+    public abstract long eventCountGet();
+
+    @Override
+    public Void call() throws InterruptedException, ExecutionException, IllegalStateException {
+        CompletableFuture retFuture = null;
+
+        do {
+            final Instant loopStartTime = Instant.now();
+            for (int i = 0; (i < eventsPerSec) && (eventCountIncrementAndGet() <= totalEvents) &&
+                    (Duration.between(loopStartTime, Instant.now()).getSeconds() < 1); i++) {
+
+                // Construct event payload
+                String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
+                String payload = String.format("%-" + messageSize + "s", val);
+                String key;
+                if (isRandomKey) {
+                    key = Integer.toString(workerID + new Random().nextInt());
```

Comment:
Don't use random as a one off like this. It is not random. You need to same it as a variable outside the loop so you can use it over and over. 
Also spotbugs should have caught this. Is it not running on this repo? If so we should enable it.

Reply:
I will do this as separate PR please.. I need to lot of stuff during random key till then i am just keeping the existing code

Reply:
Yes, add spotbugs as a separate PR. But please move the construction of random up to line 51.

Reply:
thanks.. yes , its done.